modules/users/default.nix

```diff
@@ -317,12 +317,8 @@ in
       ${concatMapStringsSep "\n" (name: ''
         u=$(id -u ${escapeShellArg name} 2> /dev/null) || true
         if [ -n "$u" ]; then
-          if [ "$u" -gt 501 ]; then
-            echo "deleting user ${name}..." >&2
-            dscl . -delete ${escapeShellArg "/Users/${name}"}
-          else
-            echo "[1;31mwarning: existing user '${name}' has unexpected uid $u, skipping...[0m" >&2
-          fi
+          echo "deleting user ${name}..." >&2
+          dscl . -delete ${escapeShellArg "/Users/${name}"}
```

Comment:
After thinking about this some more I’m wondering if we should just keep the warning here for now to unblock this without risking clobbering system users.

cc @Enzime – do you think it’s safe to remove this conditional here?

Reply:
FWIW, I think I’d be comfortable with the conservative change of allowing 350 < UID ≤ (350 + maxBuildUsers) as well as UID > 501 here. I think the idea is to avoid deleting system users.